tests/unit/maze_transformer/training/zanj/test_zanj_ht_save_load.py

```diff
@@ -0,0 +1,104 @@
+from pathlib import Path
+
+import torch
+from muutils.zanj import ZANJ
+from muutils.zanj.torchutil import (
+    ConfigMismatchException,
+    assert_model_cfg_equality,
+    assert_model_exact_equality,
+)
+
+from maze_transformer.training.config import (
+    BaseGPTConfig,
+    ConfigHolder,
+    MazeDatasetConfig,
+    TrainConfig,
+    ZanjHookedTransformer,
+)
+
+ZANJ_MODEL_CFG: ConfigHolder = ConfigHolder(
```

Comment:
Unrelated to this PR, but I think `SerializableDataclass` has some issue with its types

![Screenshot 2023-04-17 at 7 07 13 AM](https://user-images.githubusercontent.com/7400942/232467329-16d80c6d-0780-43f2-b66b-e1ec98888f7a.png)

Reply:
this might just be due to me not yet having muutils passing mypy, I'll look into it